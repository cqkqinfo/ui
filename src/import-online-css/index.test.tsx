import { transformObj } from './index';

describe('引用在线css', () => {
  it('基本', () => {
    expect(
      transformObj(`
@import "./../../remax-styles.css";
.test {
  width: 26.66667vw;
  height: 26.66667vw;
  background-color: #f00;
  font-size: 8vw;
}

:root:root {
  --adm-color-primary: #2780d9;
}
.button___xxxxxx,
.button___2hlxT {
  max-height: 13.06667vw;
  min-height: 13.06667vw;
  background: #2780d9;
  border: 1px solid #2780d9;
  border-radius: 2.66667vw;
  box-sizing: border-box;
  display: inline-flex;
  font-size: 4.8vw;
  color: #ffffff;
  justify-content: center;
  align-items: center;
  line-height: 4.8vw;
  padding: 0 1.46667vw;
  user-select: none;
  cursor: pointer;
}
.button___2hlxT:active {
  opacity: 0.7;
}
.bold___EUTWX {
  font-weight: bold;
}
.block___1_rpw {
  display: flex;
  width: 100%;
  flex: 1;
}
.attract___2tRfz {
  background: #ff9d46;
  border: 1px solid #ff9d46;
}
.ghost___3_82c {
  background-color: transparent;
  color: #2780d9;
}
.attract-ghost___7iFzL {
  color: #ff9d46;
}
.round___3_2_m {
  border-radius: 7.86667vw;
}
.loading___3nLe1 {
  pointer-events: none;
}
.disable___1ffne {
  pointer-events: none;
  background-color: #cccccc;
  border-color: #cccccc;
  color: #fff;
}
.small___qeiCp {
  max-height: 10.4vw;
  min-height: 10.4vw;
  font-size: 4vw;
  padding: 0 3.6vw;
}
.tiny___LdnO2 {
  max-height: 9.33333vw;
  min-height: 9.33333vw;
  font-size: 4.26667vw;
  padding: 0 3.6vw;
}
.action___1GQql {
  min-width: 20.26667vw;
  max-height: 9.86667vw;
  min-height: 9.86667vw;
  border-radius: 1.33333vw;
  font-size: 3.46667vw;
  width: auto;
  flex: none;
}
.default___3PkNh {
  background-color: #fff;
  border: 1px solid #e2e2e2;
  color: #999999;
}
.ghost___3_82c.default___3PkNh {
  background-color: transparent;
}
.noBorder___1U6tE {
  border: none;
  color: #2780d9;
}
.attract-noBorder___3D1-5 {
  color: #ff9d46;
}
.elderly___P2n0m.button___2hlxT {
  font-size: 5.46667vw;
}
.elderly___P2n0m.action___1GQql {
  max-height: 15.46667vw;
  background: #00b8a9;
  border-radius: 2.66667vw;
  font-size: 4.8vw;
  font-weight: bold;
  color: #ffffff;
  border-color: #00b8a9;
}
.group___3h_LF .button___2hlxT {
  width: auto;
  flex: initial;
}
.group___3h_LF .button___2hlxT:not(:first-child):not(:last-child) {
  border-radius: 0;
}
.group___3h_LF .button___2hlxT:first-child:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.group___3h_LF .button___2hlxT:last-child:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.wrap___1LAMi {
  display: flex;
  align-items: center;
  justify-content: center;
}
.icon___1oAmJ {
  width: 100%;
  height: 100%;
}

@keyframes loading-rotate___3Mvyn {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.rotate___3JMmy {
  animation: loading-rotate___3Mvyn 1s linear infinite;
}

:root:root {
  --adm-color-primary: #2780d9;
}
.part___3XIhq {
  color: #333333;
  font-size: 4vw;
  font-weight: bold;
  line-height: 4vw;
}
.block___1vZo7 {
  width: 1.2vw;
  height: 3.73333vw;
  background: #2780d9;
}
.title___3PwhH {
  margin-left: 1.6vw;
}
.today___39KY3 {
  font-size: 4vw;
  font-weight: bold;
  color: #666666;
}
.mark___3ulmI {
  font-size: 4vw;
  font-weight: bold;
  color: #d95e38;
  position: relative;
  top: -1.06667vw;
}
.elderly___1VOqu .title___3PwhH {
  font-size: 5.6vw;
  line-height: 1;
}
.elderly___1VOqu .block___1vZo7 {
  width: 1.2vw;
  height: 5.06667vw;
}
    `),
    ).toMatchSnapshot();
  });
});
