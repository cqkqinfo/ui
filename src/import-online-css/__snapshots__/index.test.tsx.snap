// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`引用在线css 基本 1`] = `
Object {
  "action1GQqlClass": Object {
    "borderRadius": "1.33333vw",
    "flex": "none",
    "fontSize": "3.46667vw",
    "maxHeight": "9.86667vw",
    "minHeight": "9.86667vw",
    "minWidth": "20.26667vw",
    "width": "auto",
  },
  "after1wn5JClass": Object {
    "display": "inline-flex",
    "marginLeft": "1.33333vw",
  },
  "attract2tRfzClass": Object {
    "background": "#ff9d46",
    "border": "1px solid #ff9d46",
  },
  "attractGhost7iFzLClass": Object {
    "color": "#ff9d46",
  },
  "attractNoBorder3D15Class": Object {
    "color": "#ff9d46",
  },
  "block1RpwClass": Object {
    "display": "flex",
    "flex": "1",
    "width": "100%",
  },
  "block1vZo7Class": Object {
    "background": "#2780d9",
    "height": "3.73333vw",
    "width": "1.2vw",
  },
  "boldEUTWXClass": Object {
    "fontWeight": "bold",
  },
  "bottom1uzPMClass": Object {
    "transform": "translateY(200vh)",
  },
  "button2hlxTActiveClass": Object {
    "opacity": "0.7",
  },
  "button2hlxTClass": Object {
    "alignItems": "center",
    "background": "#2780d9",
    "border": "1px solid #2780d9",
    "borderRadius": "2.66667vw",
    "boxSizing": "border-box",
    "color": "#ffffff",
    "cursor": "pointer",
    "display": "inline-flex",
    "fontSize": "4.8vw",
    "justifyContent": "center",
    "lineHeight": "4.8vw",
    "maxHeight": "13.06667vw",
    "minHeight": "13.06667vw",
    "padding": "0 1.46667vw",
    "userSelect": "none",
  },
  "cellChildren3pxXhClass": Object {
    "display": "flex",
    "justifyContent": "space-between",
    "textAlign": "right",
  },
  "cellChildren3pxXhtextareaClass": Object {
    "boxSizing": "border-box",
    "padding": "2.66667vw 0",
  },
  "cellChildrenLeftTPyPvClass": Object {
    "justifyContent": "flex-start",
  },
  "cellGuhZBClass": Object {
    "border": "0 solid #e2e2e2",
    "borderBottom": "1px solid #e2e2e2",
    "justifyContent": "space-between",
    "minHeight": "11.73333vw",
    "width": "100%",
  },
  "cellGuhZBLastChildClass": Object {
    "borderBottom": "none !important",
  },
  "children9PFPRClass": Object {
    "color": "#666",
    "flex": "1",
  },
  "children9PFPRPickerChildrenClass": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "9.33333vw",
    "justifyContent": "flex-end",
    "whiteSpace": "break-spaces",
    "width": "100%",
  },
  "children9PFPRpickerClass": Object {
    "width": "100%",
  },
  "colonBgNb5Class": Object {
    "display": "inline-block",
    "padding": "0 1.33333vw",
    "position": "relative",
    "top": "-0.26667vw",
  },
  "content1LwVzClass": Object {
    "height": "100vh",
    "left": "0",
    "overflow": "auto",
    "position": "absolute",
    "top": "0",
    "transition": "all 0.3s",
    "width": "100vw",
    "willChange": "transform",
  },
  "default3PkNhClass": Object {
    "backgroundColor": "#fff",
    "border": "1px solid #e2e2e2",
    "color": "#999999",
  },
  "disable1ffneClass": Object {
    "backgroundColor": "#cccccc",
    "borderColor": "#cccccc",
    "color": "#fff",
    "pointerEvents": "none",
  },
  "elderly1VOquBlock1vZo7Class": Object {
    "height": "5.06667vw",
    "width": "1.2vw",
  },
  "elderly1VOquTitle3PwhHClass": Object {
    "fontSize": "5.6vw",
    "lineHeight": "1",
  },
  "elderlyDM5G7CellGuhZBClass": Object {
    "minHeight": "15.73333vw",
  },
  "elderlyDM5G7Children9PFPRClass": Object {
    "color": "#333333",
    "fontWeight": "bold",
  },
  "elderlyDM5G7ColonBgNb5Class": Object {
    "padding": "0",
    "paddingRight": "4.26667vw",
  },
  "elderlyDM5G7Item1K29FClass": Object {
    "fontSize": "5.33333vw",
  },
  "elderlyDM5G7LabelH8Z6QClass": Object {
    "color": "#666666",
    "marginRight": "0 !important",
  },
  "elderlyP2n0mAction1GQqlClass": Object {
    "background": "#00b8a9",
    "borderColor": "#00b8a9",
    "borderRadius": "2.66667vw",
    "color": "#ffffff",
    "fontSize": "4.8vw",
    "fontWeight": "bold",
    "maxHeight": "15.46667vw",
  },
  "elderlyP2n0mButton2hlxTClass": Object {
    "fontSize": "5.46667vw",
  },
  "error1UZ47Children9PFPRClass": Object {
    "color": "#ed4e56",
  },
  "error1UZ47Class": Object {
    "color": "#ed4e56",
  },
  "error1UZ47LabelH8Z6QClass": Object {
    "color": "#ed4e56",
  },
  "error1UZ47PlaceholderClass": Object {
    "color": "#ed4e56 !important",
  },
  "ghost382cClass": Object {
    "backgroundColor": "transparent",
    "color": "#2780d9",
  },
  "ghost382cDefault3PkNhClass": Object {
    "backgroundColor": "transparent",
  },
  "group3hLFButton2hlxTClass": Object {
    "flex": "initial",
    "width": "auto",
  },
  "group3hLFButton2hlxTFirstChildNotLastChildClass": Object {
    "borderBottomRightRadius": "0",
    "borderTopRightRadius": "0",
  },
  "group3hLFButton2hlxTLastChildNotFirstChildClass": Object {
    "borderBottomLeftRadius": "0",
    "borderTopLeftRadius": "0",
  },
  "group3hLFButton2hlxTNotFirstChildNotLastChildClass": Object {
    "borderRadius": "0",
  },
  "icon1oAmJClass": Object {
    "height": "100%",
    "width": "100%",
  },
  "input2D75Class": Object {
    "backgroundColor": "transparent",
    "border": "none",
    "color": "inherit",
    "fontSize": "1em",
    "whiteSpace": "break-spaces",
    "width": "100%",
  },
  "item1K29FClass": Object {
    "alignItems": "center",
    "borderBottomColor": "transparent",
    "color": "#333",
    "display": "flex",
    "fontSize": "4vw",
    "position": "relative",
    "whiteSpace": "nowrap",
  },
  "item1K29FVertical24MTWCard2iaXgClass": Object {
    "margin": "2.66667vw 0 0",
  },
  "item1K29FVertical24MTWCard2iaXgLabelWrapYZRzXClass": Object {
    "marginBottom": "1.33333vw",
  },
  "item1K29FVertical24MTWCellGuhZBChildren9PFPRClass": Object {
    "borderTop": "1px solid #eee",
    "marginTop": "1.33333vw",
    "paddingTop": "2.66667vw",
  },
  "item1K29FVertical24MTWCellGuhZBClass": Object {
    "padding": "4vw 0",
  },
  "item1K29FVertical24MTWChildren9PFPRClass": Object {
    "width": "100%",
  },
  "item1K29FVertical24MTWClass": Object {
    "alignItems": "flex-start",
    "flexDirection": "column",
  },
  "labelH8Z6QClass": Object {
    "display": "inline-flex",
  },
  "labelWrapYZRzXClass": Object {
    "alignItems": "center",
    "display": "flex",
  },
  "left1bRpuClass": Object {
    "transform": "translateX(-200vh)",
  },
  "loading3nLe1Class": Object {
    "pointerEvents": "none",
  },
  "mark3TiN3Class": Object {
    "color": "#ea5328",
    "display": "inline-block",
    "left": "-0.5em",
    "position": "absolute",
  },
  "mark3ulmIClass": Object {
    "color": "#d95e38",
    "fontSize": "4vw",
    "fontWeight": "bold",
    "position": "relative",
    "top": "-1.06667vw",
  },
  "noBorder1U6tEClass": Object {
    "border": "none",
    "color": "#2780d9",
  },
  "part3XIhqClass": Object {
    "color": "#333333",
    "fontSize": "4vw",
    "fontWeight": "bold",
    "lineHeight": "4vw",
  },
  "right2WAfbClass": Object {
    "transform": "translateX(200vh)",
  },
  "rootRootClass": Object {
    "AdmColorPrimary": "#2780d9",
  },
  "rotate3JMmyClass": Object {
    "animation": "loading-rotate___3Mvyn 1s linear infinite",
  },
  "round32MClass": Object {
    "borderRadius": "7.86667vw",
  },
  "sheet9WewvClass": Object {
    "backgroundColor": "transparent",
    "height": "100vh",
    "left": "0",
    "opacity": "0",
    "overflow": "hidden",
    "pointerEvents": "none",
    "position": "fixed",
    "top": "0",
    "transition": "all 0.3s, z-index 0s 0.3s",
    "width": "100vw",
    "zIndex": "-1",
  },
  "show1laF7Class": Object {
    "backgroundColor": "rgba(0, 0, 0, 0.6)",
    "opacity": "1",
    "overflow": "auto",
    "pointerEvents": "auto",
    "transition": "all 0.3s",
    "zIndex": "101",
  },
  "showContent1rV2tClass": Object {
    "transform": "translateY(0) translateX(0)",
  },
  "smallQeiCpClass": Object {
    "fontSize": "4vw",
    "maxHeight": "10.4vw",
    "minHeight": "10.4vw",
    "padding": "0 3.6vw",
  },
  "testClass": Object {
    "backgroundColor": "#f00",
    "fontSize": "8vw",
    "height": "26.66667vw",
    "width": "26.66667vw",
  },
  "tinyLdnO2Class": Object {
    "fontSize": "4.26667vw",
    "maxHeight": "9.33333vw",
    "minHeight": "9.33333vw",
    "padding": "0 3.6vw",
  },
  "title3PwhHClass": Object {
    "marginLeft": "1.6vw",
  },
  "today39KY3Class": Object {
    "color": "#666666",
    "fontSize": "4vw",
    "fontWeight": "bold",
  },
  "topZIZJRClass": Object {
    "transform": "translateY(-200vh)",
  },
  "web2yhSFocusClass": Object {
    "outline": "none",
  },
  "wrap1LAMiClass": Object {
    "alignItems": "center",
    "display": "flex",
    "justifyContent": "center",
  },
}
`;
